src/backend/egl/display.rs

```diff
@@ -84,9 +93,9 @@ pub struct EGLDisplay {
 }
 
 unsafe fn select_platform_display<N: EGLNativeDisplay + 'static>(
-    native: &N,
+    native: N,
     dp_extensions: &[String],
-) -> Result<(*const c_void, &'static str), Error> {
+) -> Result<(Arc<EGLDisplayHandle>, &'static str), Error> {
```

Comment:
Would this benefit from internal ref-counting on `EGLDisplayHandle` instead of returning and using `Arc`?